NullPointerException in Bundle Activator

Having NullPointerException while starting hazelcast-3.2.1 bundle with Apache Felix 4.0.3. This results bundle in resolved state.

org.osgi.framework.BundleException: Activator start error in bundle com.hazelcast [207].
        at org.apache.felix.framework.Felix.activateBundle(Felix.java:2027)
        at org.apache.felix.framework.Felix.startBundle(Felix.java:1895)
        at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:944)
        at org.apache.felix.gogo.command.Basic.start(Basic.java:729)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
        at java.lang.reflect.Method.invoke(Method.java:597)
        at org.apache.felix.gogo.runtime.Reflective.invoke(Reflective.java:137)
        at org.apache.felix.gogo.runtime.CommandProxy.execute(CommandProxy.java:82)
        at org.apache.felix.gogo.runtime.Closure.executeCmd(Closure.java:477)
        at org.apache.felix.gogo.runtime.Closure.executeStatement(Closure.java:403)
        at org.apache.felix.gogo.runtime.Pipe.run(Pipe.java:108)
        at org.apache.felix.gogo.runtime.Closure.execute(Closure.java:183)
        at org.apache.felix.gogo.runtime.Closure.execute(Closure.java:120)
        at org.apache.felix.gogo.runtime.CommandSessionImpl.execute(CommandSessionImpl.java:89)
        at org.apache.felix.gogo.shell.Console.run(Console.java:62)
        at org.apache.felix.gogo.shell.Shell.console(Shell.java:203)
        at org.apache.felix.gogo.shell.Shell.gosh(Shell.java:128)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
        at java.lang.reflect.Method.invoke(Method.java:597)
        at org.apache.felix.gogo.runtime.Reflective.invoke(Reflective.java:137)
        at org.apache.felix.gogo.runtime.CommandProxy.execute(CommandProxy.java:82)
        at org.apache.felix.gogo.runtime.Closure.executeCmd(Closure.java:477)
        at org.apache.felix.gogo.runtime.Closure.executeStatement(Closure.java:403)
        at org.apache.felix.gogo.runtime.Pipe.run(Pipe.java:108)
        at org.apache.felix.gogo.runtime.Closure.execute(Closure.java:183)
        at org.apache.felix.gogo.runtime.Closure.execute(Closure.java:120)
        at org.apache.felix.gogo.runtime.CommandSessionImpl.execute(CommandSessionImpl.java:89)
        at org.apache.felix.shell.remote.Shell.startGogoShell(Shell.java:108)
        at org.apache.felix.shell.remote.Shell.run(Shell.java:81)
        at java.lang.Thread.run(Thread.java:662)
Caused by: java.lang.NullPointerException
        at com.hazelcast.osgi.OSGiScriptEngineManager.findFactoryCandidates(OSGiScriptEngineManager.java:285)
        at com.hazelcast.osgi.OSGiScriptEngineManager.findManagers(OSGiScriptEngineManager.java:230)
        at com.hazelcast.osgi.OSGiScriptEngineManager.<init>(OSGiScriptEngineManager.java:88)
        at com.hazelcast.osgi.Activator.start(Activator.java:45)
        at org.apache.felix.framework.util.SecureAction.startActivator(SecureAction.java:645)
        at org.apache.felix.framework.Felix.activateBundle(Felix.java:1977)
        ... 33 more

--- a/hazelcast/src/main/java/com/hazelcast/osgi/OSGiScriptEngineManager.java
+++ b/hazelcast/src/main/java/com/hazelcast/osgi/OSGiScriptEngineManager.java
@@ -285,7 +285,7 @@ public class OSGiScriptEngineManager extends ScriptEngineManager {
             if (bundle == null) {
                 continue;
             }
-            if (bundle.getSymbolicName() != null && bundle.getSymbolicName().equals("system.bundle")) {
+            if ("system.bundle".equals(bundle.getSymbolicName())) {
                 continue;
             }
             Enumeration urls = bundle.findEntries("META-INF/services", "javax.script.ScriptEngineFactory", false);

